python/pyspark/sql/dataframe.py

```diff
@@ -1697,7 +1697,19 @@ def sample(  # type: ignore[misc]
         >>> df.sample(False, fraction=1.0).count()
         10
         """
+        (withReplacement, fraction, seed) = DataFrame._prepare_augments_for_sample(
+            withReplacement, fraction, seed
+        )
+        args = [arg for arg in [withReplacement, fraction, seed] if arg is not None]
+        jdf = self._jdf.sample(*args)
+        return DataFrame(jdf, self.sparkSession)
 
+    @staticmethod
+    def _prepare_augments_for_sample(
```

Comment:
the pre-processing of `sample` augments is pretty complex, so make it a static method and reuse it in connect

Reply:
If we do need to share code between pyspark and spark connect python client, we should probably add a new module like `python-common`